### PR TITLE
ReceiptHandleProcessor message renewal strategy optimization

### LINKED ISSUE #6232: ReceiptHandleProcessor message renewal strategy optimization
ReceiptHandleProcessor messages renew for one minute at a time. It is necessary to optimize the renewal time into a retreat strategy to reduce the number of renewals.
Normally, a message will be ack within a few hundred milliseconds to a few seconds after consumption is complete. If no ack message is received within 1 minute, the client is faulty and cannot ack normally.
At this point, if you follow the fixed renewal for one minute, in extreme cases, if there are a large number of messages that need to be renewed, you will experience performance problems.                           
The current maximum renewal time is determined by the renewMaxTimeMillis configuration item. The default value is 3 hours.
There are two strategies to choose from:  
1.Set a retreat time group. 1min, 3min, 5min, 10min, 30min, 1hour. Maximum 1hour
2.The minimum is 1min, and the accumulation is carried out according to the index of 2 (unit of second). It can take the class ExponentialRetryPolicy

### PR DESCRIPTION
**Pull request by socutes**
close #6232

### CODE REVIEW

**Comment by socutes** [comment]
@xdkxlk @drpmma @zhouxinyu PTAL

File: proxy/src/main/java/org/apache/rocketmq/proxy/processor/ReceiptHandleProcessor.java [hunk 1]
```diff
@@ -175,9 +175,10 @@ protected CompletableFuture<MessageReceiptHandle> startRenewMessage(MessageRecei
                 return CompletableFuture.completedFuture(null);
             }
             if (current - messageReceiptHandle.getConsumeTimestamp() < proxyConfig.getRenewMaxTimeMillis()) {
+                RetryPolicy renewPolicy = messageReceiptHandle.getRenewStrategyPolicy();
                 CompletableFuture<AckResult> future =
                     messagingProcessor.changeInvisibleTime(context, handle, messageReceiptHandle.getMessageId(),
-                        messageReceiptHandle.getGroup(), messageReceiptHandle.getTopic(), proxyConfig.getRenewSliceTimeMillis());
+                        messageReceiptHandle.getGroup(), messageReceiptHandle.getTopic(), renewPolicy.nextDelayDuration(messageReceiptHandle.getRenewRetryTimes()));
```
Comment (xdkxlk) [inline proxy/src/main/java/org/apache/rocketmq/proxy/processor/ReceiptHandleProcessor.java:181]: We need another counter to save the renew times
The retry times will be increased when some error occur, and will reset to 0 after success

File: proxy/src/main/java/org/apache/rocketmq/proxy/common/MessageReceiptHandle.java [hunk 2]
```diff
@@ -47,6 +49,7 @@ public MessageReceiptHandle(String group, String topic, int queueId, String rece
         this.queueOffset = queueOffset;
         this.reconsumeTimes = reconsumeTimes;
         this.consumeTimestamp = receiptHandle.getRetrieveTime();
+        this.renewStrategyPolicy = new RenewStrategyPolicy();
```
Comment (xdkxlk) [inline proxy/src/main/java/org/apache/rocketmq/proxy/common/MessageReceiptHandle.java:52]: Maybe move the policy to ReceiptHandleProcessor as a static field is better

File: proxy/src/main/java/org/apache/rocketmq/proxy/processor/ReceiptHandleProcessor.java [hunk 3]
```diff
@@ -192,7 +193,7 @@ protected CompletableFuture<MessageReceiptHandle> startRenewMessage(MessageRecei
                         messageReceiptHandle.resetRenewRetryTimes();
                         resFuture.complete(messageReceiptHandle);
                     } else {
-                        log.error("renew response is not ok. result:{}, handle:{}", ackResult, messageReceiptHandle, throwable);
+                        log.error("renew response is not ok. result:{}", ackResult, messageReceiptHandle);
```
Comment (xdkxlk) [inline proxy/src/main/java/org/apache/rocketmq/proxy/processor/ReceiptHandleProcessor.java:196]: The format is not right, a `{}` is missing here

File: proxy/src/main/java/org/apache/rocketmq/proxy/processor/ReceiptHandleProcessor.java (hunk 1 above)
Comment (drpmma) [reply proxy/src/main/java/org/apache/rocketmq/proxy/processor/ReceiptHandleProcessor.java:181]: The renew times have no relationship with retry times.

File: proxy/src/main/java/org/apache/rocketmq/proxy/processor/ReceiptHandleProcessor.java (hunk 3 above)
Comment (drpmma) [reply proxy/src/main/java/org/apache/rocketmq/proxy/processor/ReceiptHandleProcessor.java:196]: The throwable argument is missing for log.error

**Review by drpmma — CHANGES_REQUESTED** [review]

File: proxy/src/main/java/org/apache/rocketmq/proxy/processor/ReceiptHandleProcessor.java (hunk 3 above)
Comment (socutes) [reply proxy/src/main/java/org/apache/rocketmq/proxy/processor/ReceiptHandleProcessor.java:196]: > The throwable argument is missing for log.error

In this judgment condition, the throwable is always empty, so no record is required.

**Comment by socutes** [comment]
@xdkxlk @drpmma PTAL, Thanks

File: proxy/src/main/java/org/apache/rocketmq/proxy/processor/ReceiptHandleProcessor.java [hunk 4]
```diff
@@ -175,9 +176,10 @@ protected CompletableFuture<MessageReceiptHandle> startRenewMessage(MessageRecei
                 return CompletableFuture.completedFuture(null);
             }
             if (current - messageReceiptHandle.getConsumeTimestamp() < proxyConfig.getRenewMaxTimeMillis()) {
+                RetryPolicy renewPolicy = new RenewStrategyPolicy();
```
Comment (xdkxlk) [inline proxy/src/main/java/org/apache/rocketmq/proxy/processor/ReceiptHandleProcessor.java:179]: Changing the RenewStrategyPolicy to `static final` can be better

Comment (socutes) [reply proxy/src/main/java/org/apache/rocketmq/proxy/processor/ReceiptHandleProcessor.java]: > Changing the RenewStrategyPolicy to `static final` can be better

Thanks，This problem has been modified.

**Review by xdkxlk — APPROVED** [review]

**Pull request merged**